internal/providers/pluginfw/products/app/resource_app.go

```diff
@@ -0,0 +1,226 @@
+	finalApp, err := waiter.Get()
+	if err != nil {
+		resp.Diagnostics.AddError("error waiting for app to be ready", err.Error())
+		return
+	}
+
+	// Store the final version of the app in state
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, finalApp, &newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (a *resourceApp) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)
+	w, diags := a.client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var app apps_tf.App
+	resp.Diagnostics.Append(req.State.Get(ctx, &app)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	appGoSdk, err := w.Apps.GetByName(ctx, app.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("failed to read app", err.Error())
+		return
+	}
+
+	var newApp apps_tf.App
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, appGoSdk, &newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (a *resourceApp) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)
+	w, diags := a.client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var app apps_tf.App
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &app)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the app
+	var appGoSdk apps.App
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, app, &appGoSdk)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	_, err := w.Apps.Update(ctx, apps.UpdateAppRequest{App: &appGoSdk, Name: app.Name.ValueString()})
+	if err != nil {
+		resp.Diagnostics.AddError("failed to update app", err.Error())
+		return
+	}
+
+	// Store the updated version of the app in state
+	var newApp apps_tf.App
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, appGoSdk, &newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newApp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (a *resourceApp) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)
+	w, diags := a.client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var app apps_tf.App
+	resp.Diagnostics.Append(req.State.Get(ctx, &app)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the app
+	_, err := w.Apps.DeleteByName(ctx, app.Name.ValueString())
```

Comment:
Going forward, should we consider using the more general `w.Apps.Delete`? I understand it achieves the same result, but with auto-generated resources might want to avoid specific identifier names. Instead, they’d probably lean on tf-to-go and the `DeleteRequest` structures to align things correctly. This way, we ensure a bit more consistency across different resources. What do you think?